v1/src/main/java/com/google/cloud/teleport/datadog/DatadogEventWriter.java

```diff
@@ -161,33 +168,63 @@ public void processElement(
       BoundedWindow window,
       @StateId(BUFFER_STATE_NAME) BagState<DatadogEvent> bufferState,
       @StateId(COUNT_STATE_NAME) ValueState<Long> countState,
+      @StateId(BUFFER_SIZE_STATE_NAME) ValueState<Long> bufferSizeState,
       @TimerId(TIME_ID_NAME) Timer timer)
       throws IOException {
 
-    Long count = MoreObjects.<Long>firstNonNull(countState.read(), 0L);
     DatadogEvent event = input.getValue();
     INPUT_COUNTER.inc();
+
+    String eventPayload = publisher.getStringPayload(event);
+    long eventPayloadSize = eventPayload.getBytes(StandardCharsets.UTF_8).length;
```

Comment:
This could probably go to a utility class? And replace here + `calculateByteSize` used in tests

Reply:
Done!